federatedscope/attack/worker_as_attacker/server_attacker.py

```diff
@@ -112,13 +114,19 @@ def run_reconstruct(self, state_list=None, sender_list=None):
                 self._reconstruct(state, sender)
 
     def callback_funcs_model_para(self, message: Message):
+        if self.is_finish:
+            return 'finish'
+
         round, sender, content = message.state, message.sender, message.content
         self.sampler.change_state(sender, 'idle')
-        # For a new round
-        if round not in self.msg_buffer['train'].keys():
-            self.msg_buffer['train'][round] = dict()
 
-        self.msg_buffer['train'][round][sender] = content
+        if round >= self.state - self.staleness_toleration:
```

Comment:
As a subclass of `Server`, is it possible to inherit this piece of code?

Reply:
`self.staleness_toleration` would be set to 0 when `asyn.use=False`, as shown in line145-147 in server.py